api/client.cpp

```diff
@@ -27,28 +27,50 @@ namespace expected = mender::common::expected;
 namespace http = mender::http;
 namespace log = mender::common::log;
 
-error::Error Client::AsyncCall(
-	http::OutgoingRequestPtr req,
-	http::ResponseHandler header_handler,
-	http::ResponseHandler body_handler) {
+http::ExpectedOutgoingRequestPtr APIRequest::WithAuthData(const auth::AuthData &auth_data) {
```

Comment:
I think the `BaseOutgoingRequest` approach turned out pretty nice. It didn't add a lot of code, and now the API is "correct at compile time".

Reply:
Yes, `= default` kind of saved the day. :grin: